scripts/cast-on-tenderly/index.js

```diff
@@ -19,63 +20,123 @@ if (!SPELL_ADDRESS) {
+                accept: 'application/json, text/plain, */*',
+                'X-Access-Key': process.env.TENDERLY_ACCESS_KEY,
+            },
+        });
+    } catch (error) {
+        console.error('makeTenderlyApiRequest error', error.response);
+        throw new Error(`tenderly request failed with: ${error.code}`);
+    }
 };
 
-const publishTenderlyTransaction = async function (forkId, transactionId) {
-    await makeTenderlyApiRequest(`/fork/${forkId}/transaction/${transactionId}/share`);
-    return `https://dashboard.tenderly.co/shared/fork/simulation/${transactionId}`;
+const createTenderlyTestnet = async function (spellName) {
+    const slug = `${spellName.replace(/\s+/g, '-').toLowerCase()}-${randomUUID()}`;
+    const response = await makeTenderlyApiRequest({
+        method: 'post',
+        path: '/testnet/container',
+        body: {
+            slug,
+            displayName: spellName,
+            description: spellName,
+            networkConfig: {
+                networkId: NETWORK_ID,
+                blockNumber: 'latest',
+                baseFeePerGas: '1',
+                chainConfig: {
+                    chainId: NETWORK_ID,
+                },
+            },
+            explorerPage: 'DISABLED',
+            syncState: true,
+        },
+    });
+    const testnetId = response.data.container.id;
+    const rpcEndpointPrivate = response.data.container.connectivityConfig.endpoints.find(
+        endpoint => endpoint.displayName === 'unlocked'
+    );
+    console.info(`tenderly testnet "${testnetId}" is created`);
+    return {
+        testnetId,
+        rpcUrlPrivate: rpcEndpointPrivate.uri,
+    };
 };
 
-const runSpell = async function () {
-    const { forkId, forkUrl, rpcUrl } = await createTenderlyFork();
-    console.info('private tenderly fork is created', forkUrl);
-
-    const provider = new ethers.providers.JsonRpcProvider(rpcUrl);
-    const signer = provider.getSigner();
+const publishTenderlyTestnet = async function (testnetId) {
+    console.info(`making tenderly testnet "${testnetId}" public...`);
+    const response = await makeTenderlyApiRequest({
+        method: 'put',
+        path: `/testnet/container/${testnetId}`,
+        body: {
+            explorerPage: 'ENABLED',
+        },
+    });
+    if (response.data?.container?.explorer_page !== 'ENABLED') {
+        throw new Error('failed to publish testnet');
+    }
+    console.info(`tenderly testnet is now public and discoverable`);
+    const rpcEndpointPublic = response.data.container.connectivityConfig.endpoints.find(
+        endpoint => endpoint.displayName === 'testnet'
+    );
+    const explorerUrlPublic = `https://dashboard.tenderly.co/explorer/vnet/${rpcEndpointPublic.id}`;
+    console.info(`public explorer url: ${explorerUrlPublic}`);
+    return { rpcUrlPrivate: rpcEndpointPublic.uri, explorerUrlPublic };
+};
 
+const makeSpellTheHat = async function (spellAddress, provider) {
```

Comment:
Nit: maybe make it a bit more readable in plain English?
```suggestion
const giveTheHatToSpell = async function (spellAddress, provider) {
```

Reply:
I guess that's the correct maker slang